src/mjolnir/ingest_transit.cc

```diff
@@ -457,58 +517,45 @@ bool write_stop_pair(Transit& tile,
         uniques.lock.unlock();
       }
 
-      bool origin_is_generated = tile_info.station_children.find({origin_stopId, currFeedPath}) !=
-                                 tile_info.station_children.end();
-      auto origin_onestop_id =
-          origin_is_generated ? origin_stopId + "_" + to_string(NodeType::kMultiUseTransitPlatform)
-                              : origin_stopId;
       stop_pair->set_origin_onestop_id(origin_onestop_id);
-
-      bool dest_is_generated = tile_info.station_children.find({dest_stopId, currFeedPath}) !=
-                               tile_info.station_children.end();
-      auto dest_onestop_id = dest_is_generated
-                                 ? dest_stopId + "_" + to_string(NodeType::kMultiUseTransitPlatform)
-                                 : dest_stopId;
       stop_pair->set_destination_onestop_id(dest_onestop_id);
 
+      stop_pair->set_destination_arrival_time(dest_stopTime.arrival_time.get_total_seconds());
+      stop_pair->set_origin_departure_time(origin_stopTime.departure_time.get_total_seconds());
+
       if (origin_is_in_tile) {
-        stop_pair->set_origin_departure_time(origin_stopTime.departure_time.get_total_seconds());
         // So we looked up the node graphid by name, the name is either the actual name of the
         // platform (track 5 or something) OR its just the name of the station (in the case that the
         // platform is generated). So when its generated we will have gotten back the graphid for
         // the parent station, not for the platform, and the generated platform in that case will be
-        // the next node after the station (we did this in write_stops)
+        // the next node after the station (we did this in write_stops); this also means that every
+        // platform which is generated has one individual station
         stop_pair->set_origin_graphid(origin_graphid_it->second +
                                       static_cast<uint64_t>(origin_is_generated));
 
-        // call function to set shape
-        float dist = get_stop_pair_dist(*origin_stop, currShape, origin_stopTime);
-        stop_pair->set_origin_dist_traveled(dist);
+        // call function to set shape, only if it has a shape_id, otherwise no need (see
+        // convert_transit)
+        if (stop_pair->has_shape_id()) {
+          float dist = get_stop_pair_dist(*origin_stop, currShape, origin_stopTime);
+          stop_pair->set_origin_dist_traveled(dist);
```

Comment:
also the dist_traveled is only interesting for convert_transit if there's a shape_id

Reply:
pull out the stop_time's dist_traveled